Source/Core/InputCommon/DynamicInputTextureConfiguration.cpp

```diff
@@ -0,0 +1,251 @@
+// Copyright 2019 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#include "InputCommon/DynamicInputTextureConfiguration.h"
+
+#include <picojson/picojson.h>
+
+#include "Common/File.h"
+#include "Common/FileUtil.h"
+#include "Common/Logging/Log.h"
+#include "Common/StringUtil.h"
+#include "Core/ConfigManager.h"
+#include "InputCommon/ControllerEmu/ControllerEmu.h"
+#include "InputCommon/ImageOperations.h"
+#include "VideoCommon/RenderBase.h"
+
+namespace InputCommon
+{
+DynamicInputTextureConfiguration::DynamicInputTextureConfiguration(const std::string& json_file)
+{
+  std::ifstream json_stream;
+  File::OpenFStream(json_stream, json_file, std::ios_base::in);
+  if (!json_stream.is_open())
+  {
+    ERROR_LOG(PAD, "Failed to load dynamic input json file '%s'", json_file.c_str());
```

Comment:
Do you have a suggestion for a logger?  It was a tossup between PAD and one of the graphics ones.  Since this is for inputs, I picked PAD

Reply:
Very good point. I'd rather use `VIDEO` (or not?) because it changes something graphics related (ie. the result is shown on screen; or _not_ when something is wrong). Guessing Screenshots/FrameDumps go to `VIDEO` for that reason.
Or we define a new one; perhaps specific to resource packs, perhaps generic enough to handle many different customization points?
I can live with either, but feel like `PAD` is the last one of them that I'd look at for a _texture_ related message.

Reply:
Yeah, I originally thought about making my own category but for such a minor feature, I didn't know if it was worth it.  I don't really mind one way or other so I'll move it to `VIDEO`.